helios-client/src/main/java/com/spotify/helios/common/descriptors/RollingOperation.java

```diff
@@ -0,0 +1,247 @@
+  }
+
+  public JobId getJobId() {
+    return jobId;
+  }
+
+  public Reason getReason() {
+    return reason;
+  }
+
+  public RolloutOptions getRolloutOptions() {
+    return rolloutOptions;
+  }
+
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  @Override
+  public boolean equals(final Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    final RollingOperation that = (RollingOperation) o;
+
+    if (jobId != null ? !jobId.equals(that.jobId) : that.jobId != null) {
+      return false;
+    }
+    if (deploymentGroupName != null ? !deploymentGroupName.equals(that.deploymentGroupName)
+                                    : that.deploymentGroupName != null) {
+      return false;
+    }
+    if (id != null ? !id.equals(that.id) : that.id != null) {
+      return false;
+    }
+    if (reason != null ? !reason.equals(that.reason) : that.reason != null) {
+      return false;
+    }
+    if (rolloutOptions != null ? !rolloutOptions.equals(that.rolloutOptions)
+                               : that.rolloutOptions != null) {
+      return false;
+    }
+
+    return true;
+  }
+
+  @Override
+  public int hashCode() {
+    int result = id != null ? id.hashCode() : 0;
+    result = 31 * result + (deploymentGroupName != null ? deploymentGroupName.hashCode() : 0);
+    result = 31 * result + (jobId != null ? jobId.hashCode() : 0);
+    result = 31 * result + (reason != null ? reason.hashCode() : 0);
+    result = 31 * result + (rolloutOptions != null ? rolloutOptions.hashCode() : 0);
+    return result;
+  }
+
+  @Override
+  public String toString() {
+    return "RollingOperation{" +
+           "id='" + id + '\'' +
+           ", deploymentGroupName=" + deploymentGroupName +
+           ", job=" + jobId +
+           ", reason=" + reason +
+           ", rolloutOptions=" + rolloutOptions +
+           '}';
+  }
+
+  public Builder toBuilder() {
+    final Builder builder = newBuilder();
+
+    return builder
+        .setJobId(jobId)
+        .setDeploymentGroupName(deploymentGroupName)
+        .setReason(reason)
+        .setRolloutOptions(rolloutOptions);
+  }
+
+  public static class Builder implements Cloneable {
+
+    private final Parameters p;
+
+    private Builder() {
+      this.p = new Parameters();
+    }
+
+    private static class Parameters implements Cloneable {
+
+      public String id;
+      public String deploymentGroupName;
+      public JobId jobId;
+      public Reason reason;
+      public RolloutOptions rolloutOptions;
+
+      private Parameters() {
+        this.id = UUID.randomUUID().toString();
+        this.deploymentGroupName = EMPTY_DEPLOYMENT_GROUP_NAME;
+        this.jobId = EMPTY_JOB_ID;
+        this.reason = Reason.MANUAL;
+        this.rolloutOptions = null;
+      }
+    }
+
+    public String getId() {
+      return p.id;
+    }
+
+    public Builder setId(final String id) {
```

Comment:
https://github.com/spotify/helios/pull/903/files#diff-64b87e65f5290ca066c979f34868c419R135

The above test currently requires me to create RollingOperations with a predictable ID, because I can't mock out calls to ZK paths composed of UUID that I don't know in advance of its creation.

I could possibly avoid this situation by breaking said test up into a few smaller ones, but it doesn't seem that bad to me to expose a knob to override the randomly generated UUID even if we don't use it much.